app/views/result/index.html.erb

```diff
@@ -30,16 +36,19 @@
           <label class="form-label" for="result_count">
             Required number of results
           </label>
-          <input class="form-control" id="result_count" name="count" type="number"></input>
+          <input value="<%= params["count"] %>" class="form-control" id="result_count" name="count" type="number"></input>
```

Comment:
given this is a rails app I would prefer to use the inbuilt helpers
```
<%= text_field_tag :count, params[:count], class: 'form-control', id: 'result_count', type: 'number' %>
```
You will need to check I have the above correct.